### PR TITLE
Automate part of TestMaxQuerySamples

### PR DESCRIPTION
**Pull request by codesome**
This was a part of https://github.com/prometheus/prometheus/pull/8121.

Since we are testing the correctness of queries elsewhere, I feel we can remove that check in `TestMaxQuerySamples`. It also makes adding more tests cases less discouraging and less tedious.

I have modified the test to only add cases where it just touches the limit while automatically producing test cases for exceeding the limit. Test cases that I removed in this PR are taken care by this.

### CODE REVIEW

File: promql/engine_test.go [hunk 1]
```diff
@@ -545,231 +545,121 @@ load 10s
...
-			End:      time.Unix(10, 0),
-			Interval: 5 * time.Second,
+			Start:      time.Unix(0, 0),
+			End:        time.Unix(10, 0),
+			Interval:   5 * time.Second,
 		},
 	}
 
+	// Adding additional test cases for exceeding limit.
+	additionalCases := make([]testCase, 0, len(cases))
```
Comment (brian-brazil) [inline promql/engine_test.go:628]: I think it'd be simpler to run the test twice below, rather than duplicating the data.

File: promql/engine_test.go [hunk 2]
```diff
@@ -545,231 +545,121 @@ load 10s
 func TestMaxQuerySamples(t *testing.T) {
 	test, err := NewTest(t, `
 load 10s
-  metric 1 2
-  bigmetric{a="1"} 1 2
-  bigmetric{a="2"} 1 2
+  metric 1+1x100
```
Comment (brian-brazil) [inline promql/engine_test.go:548]: Why did you change this?

Comment (codesome) [reply promql/engine_test.go:548]: It was required for testing @ timestamp. Will remove from this PR and add it there.

**Comment by brian-brazil** [comment]
:+1:
